src/main/java/io/strimzi/utils/KafkaVersionService.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.utils;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.JsonNode;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import org.testcontainers.shaded.org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ *  A service for querying for Kafka versions using abstract criteria such as "latest version",
+ *  the result of which may change over time (and thus, non-deterministically).
+ */
+public class KafkaVersionService {
+
+    private static final Logger LOGGER = LogManager.getLogger(KafkaVersionService.class);
+
+    private static final Pattern STRIMZI_TEST_CONTAINER_IMAGE_WITHOUT_KAFKA_VERSION = Pattern.compile("^test-container:(\\d+\\.\\d+\\.\\d+|latest)-kafka-.*$");
+    private static final String KAFKA_VERSIONS_URL_JSON = "https://raw.githubusercontent.com/strimzi/test-container-images/main/kafka_versions.yaml"; // this would be replaced by test-container-images url where json will be stored
+
+    private String jsonVersion;
+    private final List<KafkaVersion> logicalKafkaVersionEntities = new ArrayList<>();
+
+    /**
+     * Constructor of the KafkaVersionService, which invokes resolution and parsing phase. In resolution, we fetch
+     * data from @code{KAFKA_VERSIONS_URL_JSON} and then we parse that json scheme, which gives us list of @code{KafkaVersion}
+     * objects.
+     */
+    public KafkaVersionService() {
+        // scrape json schema and fill the inner list of versions
+        this.resolveAndParse();
```

Comment:
> I often work offline (yes, I'm old school)

Just turn off your notification :D. Any way I can implement some cache mechanisms... ))

Reply:
>  I often work offline (yes, I'm old school) to avoid notifications. How would that work in this case?

By current implementation, it will not work.